python/ray/tune/ray_trial_executor.py

```diff
@@ -166,13 +163,18 @@ def __init__(self,
         self._paused = {}
 
         self._trial_cleanup = _TrialCleanup()
+        self._has_cleaned_up_pgs = False
         self._reuse_actors = reuse_actors
         self._cached_actor = None
+        self._cached_pg = None
 
         self._avail_resources = Resources(cpu=0, gpu=0)
         self._committed_resources = Resources(cpu=0, gpu=0)
         self._pg_manager = PlacementGroupManager()
         self._staged_trials = set()
+        self._just_staged_trials = set()
+        self._trial_just_finished = False
+        self._trial_just_finished_before = False
```

Comment:
what's the reason for the `just_finished` variables?

Reply:
(see long explanation above)